### PR TITLE
103 Safely rescue interrupts

### LINKED ISSUE #103: Safely rescue Interrupt
If a ruby program is interrupted by `Ctrl^C` or sent a `SIGINT` signal, the `Interrupt` exception is raised, causing an ugly backtrace to be printed. Ideally, `Dry::CLI` should `rescue Interrupt` and exit with a status of 130 (see: https://tldp.org/LDP/abs/html/exitcodes.html).

### PR DESCRIPTION
**Pull request by IvanShamatov**
Resolves #103 
 

> If a ruby program is interrupted by Ctrl^C or sent a SIGINT signal, the Interrupt exception is raised, causing an ugly backtrace to be printed. Ideally, Dry::CLI should rescue Interrupt and exit with a status of 130 (see: https://tldp.org/LDP/abs/html/exitcodes.html).

Now works as expected

### CODE REVIEW

File: spec/integration/signal_trap_spec.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+RSpec.describe "SignalException handler" do
+  it "returns correct statuscode" do
+    pid1, pid2, pid15 = Array.new(3).map { Process.spawn("foo generate app SomeAppName") }
+    sleep 0.5 # time to start app
+
```
Comment (IvanShamatov) [inline spec/integration/signal_trap_spec.rb:7]: Groupped into one spec to spend less time in integration specs

File: spec/unit/dry/cli/cli_spec.rb [hunk 2]
```diff
@@ -120,4 +120,5 @@
       )
     end
   end
+
```
Comment (solnic) [inline spec/unit/dry/cli/cli_spec.rb:123]: Extra empty line here

File: lib/dry/cli/command.rb [hunk 3]
```diff
@@ -13,37 +13,20 @@ class Command
...
-
-            @options =
-              if base.superclass.instance_variable_defined?(:@options)
-                base.superclass.instance_variable_get(:@options).dup
-              else
-                []
-              end
-          end
-        end
-
```
Comment (solnic) [inline lib/dry/cli/command.rb:46]: This seems like an unrelated change or am I missing something?

Comment (IvanShamatov) [reply lib/dry/cli/command.rb:46]: Yep, @jodosha asked to make this nicer after I merged previous PR

Comment (solnic) [reply lib/dry/cli/command.rb:46]: @IvanShamatov OK in the future please use separate PRs with such unrelated improvements. Our future selves will appreciate it and it's a good habit to have in general. It makes maintenance simpler, especially when it turns out something needs to be reverted etc.

Comment (IvanShamatov) [reply lib/dry/cli/command.rb]: will do that in separate pr

**Review by solnic — APPROVED** [review]

**Pull request merged**